src/DatePicker/DateDisplay.js

```diff
@@ -128,12 +129,14 @@ class DateDisplay extends Component {
     const {prepareStyles} = this.context.muiTheme;
     const styles = getStyles(this.props, this.context, this.state);
     const year = selectedDate.getFullYear();
+    const regex = /[.,]/;
+    const delimiter = this.props.locale === 'fr' ? '.' : ',';
```

Comment:
hey dude, this technique isn't going to be scalable... I think we may just have to leave it as it was 😞 there's no way we can account for all the possible locales that a user might pass if they are providing the formatter.